enterprise/frontend/src/metabase-enterprise/tenants/EditUserStrategyModal/EditUserStrategyModal.unit.spec.tsx

```diff
@@ -46,66 +46,109 @@ describe("EditUserStrategyModal", () => {
+    ).toBeInTheDocument();
+
+    const applyButton = screen.getByRole("button", { name: /Apply/ });
+    await userEvent.click(applyButton);
 
-    await userEvent.click(select1);
-    setupPropertiesEndpoints(createMockSettings({ "use-tenants": true }));
-    await userEvent.click(await screen.findByText("Multi tenant"));
     await waitFor(async () => {
       const puts = await findRequests("PUT");
       expect(puts).toHaveLength(1);
     });
 
-    const select2 = await userStrategySelect();
-    expect(select2).toBeInTheDocument();
-    expect(select2).toHaveValue("Multi tenant");
+    const puts = await findRequests("PUT");
+    expect(puts[0].body).toEqual({ value: true });
   });
 
-  it("should handle failing to update", async () => {
-    await setup({
-      setupEndpoints: () => {
-        fetchMock.put("path:/api/setting/use-tenants", 500, {
-          overwriteRoutes: true,
-        });
-      },
+  it("reverts the selected user strategy if the setting update fails", async () => {
+    fetchMock.put(
+      "path:/api/setting/use-tenants",
+      { throws: new Error("Internal server error") },
+      { overwriteRoutes: true },
+    );
+
+    await setup();
+
+    expect(
+      await screen.findByRole("radio", {
+        name: /Single tenant/,
+        checked: true,
+      }),
+    ).toBeInTheDocument();
+
+    const multiTenantCard = screen.getByRole("radio", {
+      name: /Multi tenant/,
     });
+    await userEvent.click(multiTenantCard);
 
-    const select1 = await userStrategySelect();
-    expect(select1).toBeInTheDocument();
-    expect(select1).toHaveValue("Single tenant");
+    const applyButton = screen.getByRole("button", { name: /Apply/ });
+    await userEvent.click(applyButton);
 
-    await userEvent.click(select1);
-    await userEvent.click(await screen.findByText("Multi tenant"));
     await waitFor(() => findRequests("PUT"));
 
-    const select2 = await userStrategySelect();
-    expect(select2).toBeInTheDocument();
-    expect(select2).toHaveValue("Single tenant");
+    // Revert to single tenant as the API call failed
+    expect(
+      await screen.findByRole("radio", {
+        name: /Single tenant/,
+        checked: true,
+      }),
+    ).toBeInTheDocument();
   });
-});
 
-const userStrategySelect = () => screen.findByLabelText("User strategy");
+  it("should disable the apply button when the user strategy has not changed", async () => {
+    await setup();
+
+    expect(
+      await screen.findByRole("radio", {
+        name: /Single tenant/,
+        checked: true,
+      }),
+    ).toBeInTheDocument();
+
+    const applyButton = screen.getByRole("button", { name: /Apply/ });
+    expect(applyButton).toBeDisabled();
+
+    await userEvent.click(screen.getByRole("radio", { name: /Multi tenant/ }));
+    expect(applyButton).toBeEnabled();
+
+    await userEvent.click(screen.getByRole("radio", { name: /Single tenant/ }));
+    expect(applyButton).toBeDisabled();
+  });
+});
```

Comment:
Can you add a test that tests the cancel button that the selection would be cleared unless applied?